File: truncate.js

function truncate( str, limit ) {

    if (str != null && str.length > limit) {

        var nextspc = str.lastIndexOf(" ", limit);
        str = str.substring(0, (nextspc > 0)? nextspc : limit ).replace(/^\s+|\s+$/g,"") + "...";
    }

    return str;

    //-- hard cutoff
    //-- return (str.length > limit)? str.slice(0,limit) + "..." : str;
}

File: truncate.cs

string Truncate(string strInput, int limit)
{
    if(!String.IsNullOrEmpty(strInput) && strInput.Length > limit)
    {
        int nxtSpc = strInput.LastIndexOf(" ", limit);
        strInput = string.Format("{0}...", strInput.Substring(0,(nxtSpc > 0)? nxtSpc : limit).Trim() );
    }

    return strInput;
}